From 80d5b08d34a97db16fe12f82e9060570087ef0f0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Gonzalve?=
 <sebastien.gonzalve@aliceadsl.fr>
Date: Tue, 3 May 2022 20:26:25 +0200
Subject: [PATCH] Fix ffmpeg compilation on FC36

The new version stats:

2021-04-27 - 46dac8cf3d - lavf 59.0.100 - avformat.h
  av_find_best_stream now uses a const AVCodec ** parameter
  for the returned decoder.

So performous needs a patch to add const when needed.
---
 game/ffmpeg.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/game/ffmpeg.cc b/game/ffmpeg.cc
index 5005ed5aa..216ca6697 100644
--- a/game/ffmpeg.cc
+++ b/game/ffmpeg.cc
@@ -251,6 +251,9 @@ FFmpeg::FFmpeg(fs::path const& _filename, int mediaType) : m_filename(_filename)
 	if (err < 0) throw Error(*this, err);
 	m_formatContext->flags |= AVFMT_FLAG_GENPTS;
 	// Find a track and open the codec
+#if (LIBAVFORMAT_VERSION_INT) >= (AV_VERSION_INT(59, 0, 100))
+	const
+#endif
 	AVCodec* codec = nullptr;
 	m_streamId = av_find_best_stream(m_formatContext.get(), static_cast<AVMediaType>(mediaType), -1, -1, &codec, 0);
 	if (m_streamId < 0) throw Error(*this, m_streamId);
